libenv/unix_iface.c

```diff
@@ -704,6 +704,9 @@ static void FindV6InterfacesInfo(EvalContext *ctx, Rlist **interfaces, Rlist **h
                     xsnprintf(prefixed_ip, sizeof(prefixed_ip), IPV6_PREFIX"%s", ip->name);
                     EvalContextClassPutHard(ctx, prefixed_ip, "inventory,attribute_name=none,source=agent");
 
+                    // Add IPv6 address to sys.ip_addresses
+                    RlistAppendScalar(ips, ip->name);
```

Comment:
I think it's more correct to put inside the if below (so if the interface is ignored, don't add IP to ip_addresses), but the same could be said for the code above, so I can fix it in a follow-up.

Reply:
Cannot agree more.